### PR TITLE
Fix partial update of settings

### LINKED ISSUE #67: The UpdateDocuments method should be a partial update
Currently when doing this test:

```c#
var indexUID = "BasicDocumentsUpdateTest";
Index index = await this.client.GetOrCreateIndex(indexUID);

// Add the documents
UpdateStatus update = await index.AddDocuments(new[]
{
    new Movie { Id = "1", Name = "Batman", Genre = "Action" },
    new Movie { Id = "2", Name = "Superman" }
});

// Update the documents
update = await index.UpdateDocuments(new[] { new Movie { Id = "1", Name = "Ironman" } });

// Check the documents have been updated and added
var docs = await index.GetDocuments<Movie>();
Assert.Equal("1", docs.First().Id);
Assert.Equal("Ironman", docs.First().Name);
Assert.Equal("Action", docs.First().Genre);
Assert.Equal("2", docs.ElementAt(1).Id);
Assert.Equal("Superman", docs.ElementAt(1).Name);
docs.ElementAt(1).Genre.Should().BeNull();
```

The `Assert.Equal("Action", docs.First().Genre);` line fails because `Genre` is `null` but it should not be. The [add-or-update-documents route](https://docs.meilisearch.com/references/documents.html#add-or-update-documents) should be a **partial update**, so the value a `Genre` should still be `"Action"`.

### PR DESCRIPTION
**Pull request by gethari**
Fixes #67

### CODE REVIEW

File: src/Meilisearch/ObjectExtensions.cs [hunk 1]
```diff
@@ -21,5 +22,26 @@ public static class ObjectExtensions
                 propInfo => char.ToLowerInvariant(propInfo.Name[0]) + propInfo.Name.Substring(1),
                 propInfo => propInfo.GetValue(source, null).ToString());
         }
+
+        /// <summary>
+        /// Iterates through Ienumerable and removes null keys.
+        /// </summary>
+        /// <param name="objectToTransform">Object to transform.</param>
+        /// <returns>Returns the same IEnumerable with null keys removed. </returns>
+        public static object ConvertToObjectWithoutPropertiesWithNullValues<T>(this IEnumerable<T> objectToTransform)
```
Comment (curquiza) [inline src/Meilisearch/ObjectExtensions.cs]:
```suggestion
        public static object RemoveNullValues<T>(this IEnumerable<T> objectToTransform)
```

File: src/Meilisearch/Index.cs [hunk 2]
```diff
@@ -120,8 +120,8 @@ public async Task<UpdateStatus> UpdateDocuments<T>(IEnumerable<T> documents, str
             {
                 uri = QueryHelpers.AddQueryString(uri, new { primaryKey = primaryKey }.AsDictionary());
             }
-
-            responseMessage = await this.client.PutAsJsonAsync(uri, documents);
+            var filteredDocuments = documents.ConvertToObjectWithoutPropertiesWithNullValues();
```
Comment (curquiza) [inline src/Meilisearch/Index.cs]:
```suggestion
            var filteredDocuments = documents.RemoveNullValues();
```

**Review by curquiza — CHANGES_REQUESTED** [review]
Hello @gethari, thanks for this PR. I suggest small naming changes, but everything is ok! Thanks for fixing this!!

File: src/Meilisearch/Index.cs [hunk 3]
```diff
@@ -120,8 +120,8 @@ public async Task<UpdateStatus> UpdateDocuments<T>(IEnumerable<T> documents, str
             {
                 uri = QueryHelpers.AddQueryString(uri, new { primaryKey = primaryKey }.AsDictionary());
             }
-
-            responseMessage = await this.client.PutAsJsonAsync(uri, documents);
+            var filteredDocuments = documents.RemoveNullValues();
```
Comment (curquiza) [inline src/Meilisearch/Index.cs:124]:
```suggestion

            var filteredDocuments = documents.RemoveNullValues();
```

File: src/Meilisearch/ObjectExtensions.cs [hunk 4]
```diff
@@ -21,5 +22,26 @@ public static class ObjectExtensions
                 propInfo => char.ToLowerInvariant(propInfo.Name[0]) + propInfo.Name.Substring(1),
                 propInfo => propInfo.GetValue(source, null).ToString());
         }
+
+        /// <summary>
+        /// Iterates through Ienumerable and removes null keys.
+        /// </summary>
+        /// <param name="objectToTransform">Object to transform.</param>
+        /// <returns>Returns the same IEnumerable with null keys removed. </returns>
```
Comment (curquiza) [inline src/Meilisearch/ObjectExtensions.cs]: Can you add the documentation for the typed parameter `T`? 🙂

**Review by curquiza — CHANGES_REQUESTED** [review]

File: src/Meilisearch/ObjectExtensions.cs (hunk 4 above)
Comment (gethari) [reply src/Meilisearch/ObjectExtensions.cs]: Done @curquiza

Comment (curquiza) [reply src/Meilisearch/ObjectExtensions.cs]: Thanks!

**Comment by curquiza** [comment]
@gethari there is still this change to apply :) https://github.com/meilisearch/meilisearch-dotnet/pull/129/files#r646641400
You can just click on `Commit suggestion` button to accept my suggestion if you agree!

**Comment by gethari** [comment]
> @gethari there is still this change to apply :) https://github.com/meilisearch/meilisearch-dotnet/pull/129/files#r646641400
> You can just click on `Commit suggestion` button to accept my suggestion if you agree!

I've already made that change. Not sure why comment is not shown as outdated. Here's how it looks in [Index.cs](https://github.com/gethari/meilisearch-dotnet/blob/403311fc7c13c099a09c20d91d5802d458402731/src/Meilisearch/Index.cs#L123) now 

image

**Comment by curquiza** [comment]
The change I asked is only to add a new line after the bracket, you did not add it ;) 
That's why the comment is still here. You can accept my suggestion with the `commit suggestion` button

(this screenshot is an example)
commit_suggestion

**Review by curquiza — APPROVED** [review]
Merging!!
Thanks @gethari, it's really nice to have you as a contributor 😊 

bors merge

**Pull request merged**